### DISCUSSION TITLE
Integration with NodeJS

### DISCUSSION THREAD

**Discussion by santisoifer** [openai/whisper#637] [Q&A] [answered]
Hi there, I was looking foward to make a web app with Whisper, but when I started seraching for information about how could I integrate NodeJs and Whisper and I didn't find anyone who had the same question, so there wasn't an answer. Is there any way to make that posible? Or I have to integrate Python in my web? Thank you.

**Comment by mayeaux** [comment, marked as answer]
You can check out my app here https://github.com/mayeaux/generate-subtitles

It's built using Nodejs and Whisper

**Reply by santisoifer** [reply]
Thank you very much! 😁

**Reply by Illyism** [reply]
I made a simple version as an OpenAI wrapper https://github.com/Illyism/whisper-docker

**Comment by Goddard** [comment]
If it is possible to load the model with onnx then it should be some what trivial to get it working.

For example : https://github.com/Fhrozen/whisper_onnx
And : https://www.w3.org/2020/06/machine-learning-workshop/talks/onnx_js_a_javascript_library_to_run_onnx_models_in_browsers_and_node_js.html

**Comment by FerLuisxd** [comment]
I'm currently looking for a solution too, other users pointed to use the CLI or their API.
But I think it's best to be able to call the model on memory rather than invoking it every time you want to do a new transcript (which takes a lot of time)

**Reply by juzarantricreole** [reply]
Did you find any way to integrate with nodejs?

**Reply by santisoifer** [reply]
Not yet. I founded many people doing it using the CLI but I don't know how to do it. However I think some answers here may help you. Let me know if I'm right!

**Reply by FerLuisxd** [reply]
Hey guys! I think one solution could be python binding (haven't tried it yet but it looks really promising!).
https://github.com/hmenyus/node-calls-python

**Reply by ChetanXpro** [reply]
hey you could try this  npm package https://www.npmjs.com/package/nodejs-whisper , nodejs binding for model,
i am using this to get word level timestamps for one of my project

**Reply by ayoadeoye1** [reply]
it's full of bug. may work fine for your personal purpose

**Comment by juzarantricreole** [comment]
I think we can use nodejs opnai library. it provide transcription mechanism

On Thu, Jun 15, 2023 at 4:21 PM Santi ***@***.***> wrote:

> Not yet. I founded many people doing it using the CLI but I don't know how
> to do it. However I think some answers here may help you. Let me know if
> I'm right!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/openai/whisper/discussions/637#discussioncomment-6184906>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BAKWQTYIZYGSVVFOV7QB7JDXLLSJXANCNFSM6AAAAAASTULISY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Regards
--


[image: user]
Juzar Antri
JR. SOFTWARE ENGINEER
[image: msg] ***@***.***> [image: linkedin]
<https://www.linkedin.com/in/juzar-antri-10537b212/> [image: github]
<https://github.com/juzarantri>

[image: logo] <https://www.creolestudios.com/>
[image: web-apps]
<https://www.creolestudios.com/our-services/ecommerce-web-development/> Web
Apps <https://www.creolestudios.com/our-services/ecommerce-web-development/>
<https://www.creolestudios.com/our-services/ecommerce-web-development/>
[image: cloud-apps]
<https://www.creolestudios.com/our-services/developing-cloud-based-systems-apps/>
Cloud
Apps
<https://www.creolestudios.com/our-services/developing-cloud-based-systems-apps/>
<https://www.creolestudios.com/our-services/developing-cloud-based-systems-apps/>
[image: mobile-apps]
<https://www.creolestudios.com/our-services/mobile-app-development/> Mobile
Apps <https://www.creolestudios.com/our-services/mobile-app-development/>
<https://www.creolestudios.com/our-services/mobile-app-development/>
[image: resources] <https://www.creolestudios.com/> Resources for Hire
<https://www.creolestudios.com/our-services/dedicated-resource-hiring/>
<https://www.creolestudios.com/our-services/dedicated-resource-hiring/>
*Disclaimer:* Information transmitted/written in this e-mail is proprietary
to the sender and is intended for exclusive use by the individual or entity
to which it is addressed by the sender (subject to his duties/delegation
power) and may contain information that is privileged, confidential or
exempt from disclosure under applicable law and shall not attach any
liability on the originator. Any use, distribution, transmission,
forwarding, printing, copying or dissemination of this information in any
way or in any manner is strictly prohibited. All dispute if any are subject
to the jurisdiction at the place where the sender resides. If you have
reason to believe that you are not the intended recipient of this
communication, please contact the sender immediately. No responsibility
whatsoever it may be for loss or damage arising from the use of the
information transmitted by this email including damage from virus on the
sender.

**Reply by ChetanXpro** [reply]
hey you could try this npm package https://www.npmjs.com/package/nodejs-whisper , nodejs binding for model,
i am using this to get word level timestamps for one of my project
